File: backend/sharing/src/main/java/com/example/sharing/dto/UserDto.java
package com.example.sharing.dto;

import java.math.BigDecimal;

public class UserDto {
    private Long userId;
    private String name;
    private String email;
    private String telegramId;
    private String profileText;
    private BigDecimal rating;
    private BigDecimal currencyBalance;

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getTelegramId() {
        return telegramId;
    }

    public void setTelegramId(String telegramId) {
        this.telegramId = telegramId;
    }

    public String getProfileText() {
        return profileText;
    }

    public void setProfileText(String profileText) {
        this.profileText = profileText;
    }

    public BigDecimal getRating() {
        return rating;
    }

    public void setRating(BigDecimal rating) {
        this.rating = rating;
    }

    public BigDecimal getCurrencyBalance() {
        return currencyBalance;
    }

    public void setCurrencyBalance(BigDecimal currencyBalance) {
        this.currencyBalance = currencyBalance;
    }
}

File: frontend/sharing-react-app/src/dto/UserDto.ts
// src/dto/UserDto.ts

// Описание интерфейса для DTO пользователя
export interface UserDto {
    userId: number | null; // Идентификатор пользователя, может быть null для новых пользователей
    name: string; // Имя пользователя
    email: string; // Email пользователя
    telegramId: string; // Telegram ID пользователя
    profileText: string; // Текст профиля пользователя
    rating: number; // Рейтинг пользователя
    currencyBalance: number; // Баланс внутренней валюты пользователя
  }
  